src/index.js

```diff
@@ -104,41 +124,202 @@ class ReactPlayerLoader extends React.Component {
-  componentWillUnmount() {
-    if (!this.mounted_) {
+  disposePlayer() {
+
+    // Nothing to dispose.
+    if (!this.player) {
       return;
     }
 
-    this.mounted_ = false;
+    // Dispose an in-page player.
+    if (this.player.dispose) {
+      this.player.dispose();
 
-    if (!this.player) {
-      return;
+    // Dispose an iframe player.
+    } else if (this.player.parentNode) {
+      this.player.parentNode.removeChild(this.player);
     }
 
-    disposePlayer(this.player);
+    // Null out the player reference.
     this.player = null;
   }
 
   /**
-   * Renders the player
+   * Update the player based on changes to certain props that do not require
+   * a full player reload.
+   *
+   * @param {Object} changes
+   *        An object. The keys of this object are the props that changed.
+   */
+  updatePlayer(changes) {
+
+    // No player exists or player is disposed.
+    if (!this.player || !this.player.el()) {
+      return;
+    }
+
+    const findPlaylistItemIndex = (items) => {
+      if (Array.isArray(items) && this.props.playlistVideoId) {
+        for (let i = 0; i < items.length; i++) {
+          const item = items[i];
+
+          if (
+            item.id === this.props.playlistVideoId ||
+            `ref:${item.referenceId}` === this.props.playlistVideoId
+          ) {
+            return i;
+          }
+        }
+      }
+      return -1;
+    };
+
+    const catalogCallback = (requestType) => {
+      return (err, data) => {
+        if (err) {
+          logError(err);
+          return;
+        }
+
+        // If the playlistVideoId changed and this is a playlist request, we
+        // need to search through the playlist items to find the correct
+        // starting index.
+        if (requestType === 'playlist' && changes.playlistVideoId) {
+          const i = findPlaylistItemIndex(data);
+
+          if (i > -1) {
+            data.startingIndex = i;
+          }
+        }
+
+        this.player.catalog.load(data);
+      };
+    };
+
+    let catalogParams;
+
+    if (this.player.usingPlugin('catalog')) {
+
+      // There is a new catalog sequence request. This takes precedence over
+      // other catalog updates because it is a different call.
+      if (changes.catalogSequence && this.props.catalogSequence) {
+        this.player.catalog.getLazySequence(this.props.catalogSequence, catalogCallback('sequence'), this.props.adConfigId);
+        return;
+      }
+
+      if (changes.videoId && this.props.videoId) {
+        catalogParams = {
+          type: 'video',
+          id: this.props.videoId
+        };
+      } else if (changes.playlistId && this.props.playlistId) {
+        catalogParams = {
+          type: 'playlist',
+          id: this.props.playlistId
+        };
+      } else if (changes.catalogSearch && this.props.catalogSearch) {
+        catalogParams = {
+          type: 'search',
+          q: this.props.catalogSearch
+        };
+      }
+    }
+
+    if (catalogParams) {
```

Comment:
Should this be in the block that makes sure we are using the catalog plugin as well?

Reply:
It doesn't need to be because `catalogParams` will be `undefined`. I'll add a comment.